benchmark/benchmark-atomic.cpp

```diff
@@ -0,0 +1,223 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-24, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+// RAJA/RAJA.hpp cannot be included here because the include logic will
+// default all atomic implementations to a desul backend.
+#include "RAJA/policy/loop/policy.hpp"
+#include "RAJA/policy/openmp/atomic.hpp"
+#include "RAJA/RAJA.hpp"
+#include "RAJA/policy/openmp/policy.hpp"
+// Conditional compilation for CUDA benchmarks.
+#include <type_traits>
+#include <iostream>
+#include <sstream>
+
+/// This helper template is used to deduce if device memory allocations are necessary
+/// inside the body of the benchmark, using the type of the execution policy.
+template<typename>
+struct IsGPU : public std::false_type {};
+
+#if defined RAJA_ENABLE_CUDA
+#include "RAJA/policy/cuda.hpp"
+#include "RAJA/policy/cuda/atomic.hpp"
+
+template<int M>
+struct IsGPU<RAJA::cuda_exec<M>> : public std::true_type {};
+
+template<int BLOCK_SZ>
+struct ExecPolicyGPU {
+    using policy = RAJA::cuda_exec<BLOCK_SZ>;
```

Comment:
```suggestion
    using policy = RAJA::cuda_exec<BLOCK_SZ, true /*asynchronous*/>;
```

By default, RAJA exec policies are synchronous. To make them asynchronous, you add a template parameter with a bool true value.